packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/GoogleMapsGroundOverlayControllerTests.m

```diff
@@ -191,4 +185,100 @@ - (void)testUpdatingGroundOverlayWithBounds {
   XCTAssertEqual(convertedPlatformGroundOverlay.zIndex, platformGroundOverlay.zIndex);
 }
 
+- (void)testUpdateGroundOverlaySetsVisibilityLast {
+  PropertyOrderValidatingGroundOverlay *groundOverlay =
+      [[PropertyOrderValidatingGroundOverlay alloc] init];
+  [FGMGroundOverlayController
+            updateGroundOverlay:groundOverlay
+      fromPlatformGroundOverlay:
+          [FGMPlatformGroundOverlay
+              makeWithGroundOverlayId:@"groundOverlay"
+                                image:[FGMPlatformBitmap
+                                          makeWithBitmap:[FGMPlatformBitmapDefaultMarker
+                                                             makeWithHue:@0]]
+                             position:[FGMPlatformLatLng makeWithLatitude:0 longitude:0]
+                               bounds:[FGMPlatformLatLngBounds
+                                          makeWithNortheast:[FGMPlatformLatLng
+                                                                makeWithLatitude:54.4816
+                                                                       longitude:5.1791]
+                                                  southwest:[FGMPlatformLatLng
+                                                                makeWithLatitude:52.4816
+                                                                       longitude:3.1791]]
+                               anchor:[FGMPlatformPoint makeWithX:0.5 y:0.5]
+                         transparency:0.5
+                              bearing:65.0
+                               zIndex:2.0
+                              visible:YES
+                            clickable:YES
+                            zoomLevel:nil]
+                    withMapView:[GoogleMapsGroundOverlayControllerTests mapView]
+                      registrar:nil
+                    screenScale:1.0
+                    usingBounds:YES];
+  XCTAssertTrue(groundOverlay.hasSetMap);
+}
+
+/// Returns a simple map view to add map objects to.
++ (GMSMapView *)mapView {
+  GMSMapViewOptions *mapViewOptions = [[GMSMapViewOptions alloc] init];
+  mapViewOptions.frame = CGRectMake(0, 0, 100, 100);
+  mapViewOptions.camera = [[GMSCameraPosition alloc] initWithLatitude:0 longitude:0 zoom:0];
+  return [[PartiallyMockedMapView alloc] initWithOptions:mapViewOptions];
+}
+
+@end
+
+@implementation PropertyOrderValidatingGroundOverlay
```

Comment:
I considered doing something sketchy with overriding core Obj-C selector handling methods to catch all property setters at once, instead of having boilerplate like this for every map object, but since that would have had to be rewritten to something like this to convert the tests to Swift I decided doing this was the better option.